src/components/ToDoForm/ToDoForm.jsx

```diff
@@ -1,33 +1,30 @@
-import PropTypes from 'prop-types'
-import { useEffect, useState } from 'react'
+import PropTypes from "prop-types";
+import { useState } from "react";
 
 const ToDoForm = ({ tasks, setTasks, editingTask, saveTask, cancelEdit }) => {
-  const [taskInput, setTaskInput] = useState('')
-
-  useEffect(() => {
-    if (editingTask !== null) {
-      setTaskInput(editingTask)
-    }
-  }, [editingTask])
```

Comment:
this is the most important thing to highlight in your solution. Here , this **shouldn't** be an `useEffect()`, remember, 

### 🕷️ _With great power, comes great responsibility_


![Screenshot 2024-09-18 at 18 11 56](https://github.com/user-attachments/assets/4e355561-f66a-4327-8139-ad1ac80c1c55)

this is the **perfect** example of what i was trying to express there: the current task for this component , comes from internal interaction in the app, not from and external sync. 
This is an infered state, this comes from the father component. do **NOT** `useEffect()` it :)

Reply:
Thanks for the feedback and the suggestions!